src/inference/include/openvino/runtime/core.hpp

```diff
@@ -507,21 +507,57 @@ class OPENVINO_RUNTIME_API Core {
      */
     Any get_property(const std::string& device_name, const std::string& name) const;
 
+    /**
+     * @brief Gets properties dedicated to device behaviour.
+     *
+     * The method is targeted to extract information which can be set via set_property method.
+     *
+     * @param device_name  - A name of a device to get a properties value.
+     * @param name  - property name.
+     * @param arguments  - additional arguments to get property
+     * @return Value of property corresponding to property name.
+     */
+    Any get_property(const std::string& device_name, const std::string& name, const AnyMap& arguments) const;
+
     /**
      * @brief Gets properties dedicated to device behaviour.
      *
      * The method is needed to request common device or system properties.
      * It can be device name, temperature, other devices-specific values.
      *
      * @tparam T - type of returned value
+     * @tparam M - property mutability
      * @param deviceName  - A name of a device to get a properties value.
      * @param property  - property object.
      * @return Property value.
      */
     template <typename T, PropertyMutability M>
     T get_property(const std::string& deviceName, const ov::Property<T, M>& property) const {
         auto to = Any::make<T>();
-        get_property(deviceName, property.name(), to);
+        get_property(deviceName, property.name(), {}, to);
+        return to.template as<T>();
+    }
+
+    /**
+     * @brief Gets properties dedicated to device behaviour.
+     *
+     * The method is needed to request common device or system properties.
+     * It can be device name, temperature, other devices-specific values.
+     *
+     * @tparam T - type of returned value
+     * @tparam M - property mutability
+     * @tparam Args - set of additional arguments ended with property object variable
+     * @param deviceName  - A name of a device to get a properties value.
+     * @param property  - property object.
+     * @param args - Optional pack of pairs: (argument name, argument value) ended with property object
+     * @return Property value.
+     */
+    template <typename T, PropertyMutability M, typename... Args>
+    util::EnableIfAllStringAny<T, Args...> get_property(const std::string& deviceName,
```

Comment:
It looks like this method doesn't work as expected, I tried call it as follow:
```
    ASSERT_NO_THROW(p = ie.get_property(deviceName, ov::max_batch_size,
                                                    std::make_pair(ov::hint::model_ptr.name(), simpleNetwork),
                                                    std::make_pair(ov::streams::num.name(), n_streams),
                                                    std::make_pair(ov::intel_gpu::hint::available_device_mem.name(), available_device_mem_size)));
```
And the optional map is empty in `Any Core::get_property(const std::string& deviceName, const std::string& name, const AnyMap& arguments)` method